api/server/router/system/system_routes.go

```diff
@@ -94,8 +94,10 @@ func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter,
 		return err
 	}
 
+	version := httputils.VersionFromContext(ctx)
+
 	var getContainers, getImages, getVolumes, getBuildCache bool
-	if typeStrs, ok := r.Form["type"]; !ok {
+	if typeStrs, ok := r.Form["type"]; !ok || versions.LessThan(version, "1.42") {
```

Comment:
This is hard to read (for me).
Can you split this `if` to multiple lines? I think that will help.

So...

```go
types, ok := r.Form["type"]
if versions.LesstThan(version, "1.42") || !ok {
 //
} else {
 //
}
```